connectors/sources/outlook.py

```diff
@@ -0,0 +1,992 @@
+                )
+        except Exception as exception:
+            raise SSLFailed(
+                f"Something went wrong while verifying SSL certificate. Error: {exception}"
+            ) from exception
+
+
+class ExchangeUsers:
+    """Fetch users from Exchange Active Directory"""
+
+    def __init__(self, ad_server, domain, exchange_server, user, password, ssl_enabled):
+        self.ad_server = Server(host=ad_server)
+        self.domain = domain
+        self.exchange_server = exchange_server
+        self.user = user
+        self.password = password
+        self.ssl_enabled = ssl_enabled
+
+    @cached_property
+    def _create_connection(self):
+        return Connection(
+            server=self.ad_server,
+            user=self.user,
+            password=self.password,
+            client_strategy=SAFE_SYNC,
+            auto_bind=True,  # pyright: ignore
+        )
+
+    async def close(self):
+        pass
+
+    def _fetch_normal_users(self, search_query):
+        try:
+            has_value_for_normal_users, _, response, _ = self._create_connection.search(
+                search_query,
+                SEARCH_FILTER_FOR_NORMAL_USERS,
+                attributes=["mail"],
+            )
+
+            if not has_value_for_normal_users:
+                raise UsersFetchFailed(
+                    "Error while fetching users from Exchange Active Directory."
+                )
+
+            for user in response:
+                yield user
+
+        except Exception as e:
+            raise UsersFetchFailed(
+                f"Something went wrong while fetching users. Error: {e}"
+            ) from e
+
+    def _fetch_admin_users(self, search_query):
+        try:
+            (
+                has_value_for_admin_users,
+                _,
+                response_for_admin,
+                _,
+            ) = self._create_connection.search(
+                search_query,
+                SEARCH_FILTER_FOR_ADMIN,
+                attributes=["mail"],
+            )
+
+            if not has_value_for_admin_users:
+                raise UsersFetchFailed(
+                    "Error while fetching users from Exchange Active Directory."
+                )
+
+            for user in response_for_admin:
+                yield user
+        except Exception as e:
+            raise UsersFetchFailed(
+                f"Something went wrong while fetching users. Error: {e}"
+            ) from e
+
+    async def get_users(self):
+        warnings.filterwarnings("default")
+        ldap_domain_name_list = ["DC=" + domain for domain in self.domain.split(".")]
+        search_query = ",".join(ldap_domain_name_list)
+
+        for user in self._fetch_normal_users(search_query=search_query):
+            yield user
+
+        for user in self._fetch_admin_users(search_query=search_query):
+            yield user
+
+    async def get_user_accounts(self):
+        global global_dns_name
```

Comment:
Is this newly added? Why you need a global variable here?

Reply:
Yes, this is newly added for SSL stuffs.